Mage.Sets/src/mage/cards/t/TheFourthDoctor.java

```diff
@@ -0,0 +1,243 @@
+        this.setRuleVisible(false);
+    }
+
+    private TheFourthDoctorTriggeredAbility(final TheFourthDoctorTriggeredAbility ability) {
+        super(ability);
+    }
+
+    @Override
+    public TheFourthDoctorTriggeredAbility copy() {
+        return new TheFourthDoctorTriggeredAbility(this); 
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.SPELL_CAST
+            || event.getType() == GameEvent.EventType.LAND_PLAYED;
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        // returns true if a card was played via this card's ability 
+        // (i.e. same identifier and same approving object)
+        if (event.hasApprovingIdentifier(MageIdentifier.TheFourthDoctorWatcher)) {
+            return event
+                    .getApprovingObject()
+                    .getApprovingAbility()
+                    .getSourceId()
+                    .equals(this.getSourceId());
+        }
+        return false;
+    }
+}
+
+// adapted from OnceEachTurnCastWatcher
+class TheFourthDoctorWatcher extends Watcher {
+
+    // we store a map of playerIds linked to a set of used approving objects.
+    private final Map<UUID, Set<MageObjectReference>> usedFrom = new HashMap<>();
+
+    TheFourthDoctorWatcher() {
+        super(WatcherScope.GAME);
+    }
+
+    @Override
+    public void watch(GameEvent event, Game game) {
+        if ((event.getType() == GameEvent.EventType.SPELL_CAST
+                ||event.getType() == GameEvent.EventType.LAND_PLAYED)
+                && event.getPlayerId()!= null
+                && event.hasApprovingIdentifier(MageIdentifier.TheFourthDoctorWatcher)) {
+            usedFrom.computeIfAbsent(event.getPlayerId(), k -> new HashSet<>())
+                    .add(event.getApprovingObject().getApprovingMageObjectReference());
+
+        }
+    }
+
+    @Override
+    public void reset() {
+        super.reset();
+        usedFrom.clear();
+    }
+
+    boolean isAbilityUsed(UUID playerId, MageObjectReference mor) {
+        return usedFrom.getOrDefault(playerId, Collections.emptySet()).contains(mor);
+    }
+
+    public static Hint getHint() {
+        return TheFourthDoctorHint.instance;
+    }
+
+}
+
+enum TheFourthDoctorHint implements Hint {
+    instance;
+
+    @Override
+    public String getText(Game game, Ability ability) {
+        TheFourthDoctorWatcher watcher = game.getState().getWatcher(TheFourthDoctorWatcher.class);
+        if (watcher != null) {
+            boolean used = watcher.isAbilityUsed(ability.getControllerId(), new MageObjectReference(ability.getSourceId(), game));
+            if (used) {
+                Player player = game.getPlayer(ability.getControllerId());
+                if (player != null) {
+                    return "A historic land has been played or a historic spell has been cast by " + player.getLogName() + " with The Fourth Doctor this turn.";
```

Comment:
Must invert text logic and use ConditionHint instead. No need in player name.

good/bad icon - You can play historic land or cast historic spell from top of your library